### ISSUE TITLE
Invalid inline schema defined in oneOf according to spec

### ISSUE THREAD

**Issue by affanshahid** [juhaku/utoipa#346]
Trying to use OpenAPI Generator CLI for my API which has some enums with type tag. I'm getting the following error:

```
WARN  o.o.codegen.DefaultCodegen - Invalid inline schema defined in oneOf/anyOf in 'CreateQuestionData'. Per the OpenApi spec, for this case when a composed schema defines a discriminator, the oneOf/anyOf schemas must use $ref. Change this inline definition to a $ref definition
```

This leads to invalid type generation for enums. I am assuming this is because utoipa generates an inline schema for enums whereas according to the spec it should use $refs?

Any workarounds?

**Comment by juhaku** [comment]
Not sure about  this. Probably depending on generation library itself. According to spec enums are allowed to have inlined schemas as well or references. Way I have validated the spec is basically checked it with editor.swagger.io to see whether it gives any lint errors. 

Since the rust types are so flexible it also depends on how you have defined your enum. For tagged enums with `serde(tag = ...)` The enums are most likely necessary to be inlined in order to create the schema serde json would output. Other way would be to start restricting what kind of types users can actually define. But I would not like to choose restricting unless it is absolutely necessary.

But more context to this would be appreciated, like what kind of enum you have defined, and what is the code generation library you have used here? Perhaps I will find time to investigate this in future, but I cannot promise anything.

**Comment by juhaku** [comment]
@affanshahid I commented on another issue related to this discriminator thing. I'd like some ideas of regarding how to solve this, see https://github.com/juhaku/utoipa/issues/621#issuecomment-2359369248

**Comment by juhaku** [comment]
Going to be fixed in #1059 as explained here https://github.com/juhaku/utoipa/issues/617#issuecomment-2359372341
